### ISSUE TITLE
Bug: UseNpmCi broken in build-nodejs18.x:1.85-x86_64

### ISSUE THREAD

**Issue by snebjorn** [aws/aws-sam-build-images#90]
### Description:

Using UseNpmCi in `build-nodejs18.x:1.85-x86_64`
```yml
Metadata:
    BuildMethod: esbuild
    BuildProperties:
        UseNpmCi: True
```
causes this error

```
$ sam build --template template.yaml --use-container

Build Failed
Error: NodejsNpmEsbuildBuilder:EsbuildBundle - Esbuild Failed: ✘ [ERROR] Invalid build flag: "--use-npm-ci"
```

When using an older version it works. public.ecr.aws/sam/build-nodejs18.x:1.84-x86_64

```
$ sam build --template template.yaml --use-container --build-image public.ecr.aws/sam/build-nodejs18.x:1.84-x86_64
```

### Steps to reproduce:

Use UseNpmCi and build using public.ecr.aws/sam/build-nodejs18.x:1.85.0-x86_64 which is latest as of writing

### Observed result:
```
Fetching public.ecr.aws/sam/build-nodejs18.x:latest-x86_64 Docker container image......

...OMITTED...

NodejsNpmEsbuildBuilder:CopySource succeeded
Running NodejsNpmEsbuildBuilder:NpmInstall
NODEJS installing in: /tmp/samcli/scratch
executing NPM: ['npm', 'install', '-q', '--no-audit', '--no-save', '--unsafe-perm', '--production']
NodejsNpmEsbuildBuilder:NpmInstall succeeded
Running NodejsNpmEsbuildBuilder:EsbuildBundle
NODEJS building ['/tmp/samcli/scratch/handler.ts'] using esbuild to /tmp/samcli/artifacts
Using the following default args: ['--bundle', '--platform=node', '--outdir=/tmp/samcli/artifacts', '--format=cjs']
Configuring the parameter 'use_npm_ci=True'
Configuring the parameter 'minify=True'
Configuring the parameter 'target=es2020'
Configuring the parameter 'sourcemap=True'
Found the following args in the config: ['--use-npm-ci', '--minify', '--target=es2020', '--sourcemap']
checking for esbuild in: ['/tmp/samcli/scratch/node_modules/.bin']
potential esbuild binaries: ['/tmp/samcli/scratch/node_modules/.bin/esbuild']
executing Esbuild: ['/tmp/samcli/scratch/node_modules/.bin/esbuild', 'handler.ts', '--bundle', '--platform=node', '--outdir=/tmp/samcli/artifacts', '--format=cjs', '--use-npm-ci', '--minify', '--target=es2020', '--sourcemap']
NodejsNpmEsbuildBuilder:EsbuildBundle failed
Traceback (most recent call last):
  File "/usr/local/opt/sam-cli/lib64/python3.7/site-packages/aws_lambda_builders/workflows/nodejs_npm_esbuild/actions.py", line 98, in execute
    self._subprocess_esbuild.run(args, cwd=self._working_directory)
  File "/usr/local/opt/sam-cli/lib64/python3.7/site-packages/aws_lambda_builders/workflows/nodejs_npm_esbuild/esbuild.py", line 97, in run
    raise EsbuildExecutionError(message=err.decode("utf8").strip())
aws_lambda_builders.workflows.nodejs_npm_esbuild.exceptions.EsbuildExecutionError: Esbuild Failed: ✘ [ERROR] Invalid build flag: "--use-npm-ci"
```

### Expected result:

Expected it to build.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Windows 11
2. If using SAM CLI, `sam --version`: SAM CLI, version 1.84.0
3. AWS region: not relevant

**Comment by jfuss** [comment]
A recent change in AWS Lambda Builders seems like the root cause of the issue. PR for the patch: https://github.com/aws/aws-lambda-builders/pull/510

**Comment by mildaniel** [comment]
Closing as this bug has now been fixed and released in version 1.86.0 of the CLI. Should anything new arise, please open a new issue.
